### PR TITLE
STAC fixes

### PR DESCRIPTION
**Pull request by alexgleith**
Implement a `numberMatched` and `context/matched` count of datasets that match the query.

Also fix a bug with queries with limit = 0.

Tests implemented ;-)

### CODE REVIEW

**Comment by jeremyh** [comment]
My reading was that `matched` was the total count of matching records in the database for the query, `returned` was the current page of results. (from [here](https://stacspec.org/STAC-api.html#operation/getFeatures)).

We're currently adding `matched` when we have a summary record that has a count already (where it's cheap and doesn't need a full db scan) https://github.com/opendatacube/datacube-explorer/blob/003cf56947749b02ed747ec4be50675bd8eacb2f/cubedash/_stac.py#L365-L368

**Comment by jeremyh** [comment]
... though that's only for a full-collection query. In theory we could do it for others that match an existing summary.

To do it properly we'll need to add an extra `count()` query without a limit, to get the full, exact matching record count.

**Review by jeremyh — COMMENTED** [review]
Looks nice! ... and you're quick too.

I think we need all of the `where` clauses to match, otherwise the counts wont match the results? It's missing the the `dataset_ids` and `requires_geometry` logic from `search_items`.

**Comment by alexgleith** [comment]
> I think we need all of the `where` clauses to match, otherwise the counts wont match the results? It's missing the the `dataset_ids` and `requires_geometry` logic from `search_items`.

Yeah, you're right. Fixed, I think. I guess I should tweak the commits now...

File: cubedash/summary/_stores.py [hunk 1]
```diff
@@ -815,6 +777,95 @@ def search_items(
...
+            dataset_ids=dataset_ids,
+            require_geometry=require_geometry,
+        )
+
+        result = self._engine.execute(query).fetchall()
+
+        if len(result) != 0:
+            return result[0][0]
+        else:
+            return 0
```
Comment (jeremyh) [inline cubedash/summary/_stores.py:809]: It's only cosmetic, but sqlalchemy has a neat `engine.execute(query).scalar()` function to query a single number succinctly.

**Review by jeremyh — APPROVED** [review]
:+1: looks good

File: cubedash/summary/_stores.py (hunk 1 above)
Comment (alexgleith) [reply cubedash/summary/_stores.py:809]: Aha, thanks.

**Pull request merged**